evmrpc/tx_test.go

```diff
@@ -0,0 +1,50 @@
+package evmrpc
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/sei-protocol/sei-chain/x/evm/types"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGetTxReceipt(t *testing.T) {
+	types.RegisterInterfaces(EncodingConfig.InterfaceRegistry)
+
+	require.Nil(t, EVMKeeper.SetReceipt(Ctx, common.HexToHash("0x1234567890123456789012345678901234567890123456789012345678901234"), &types.Receipt{
+		From:              "0x123456789012345678902345678901234567890",
+		To:                "0x123456789012345678902345678901234567890",
+		TransactionIndex:  0,
+		BlockNumber:       8,
+		TxType:            1,
+		ContractAddress:   "0x123456789012345678902345678901234567890",
+		CumulativeGasUsed: 123,
+		TxHashHex:         "0x123456789012345678902345678901234567890123456789012345678901234",
+		GasUsed:           55,
+		Status:            0,
+		EffectiveGasPrice: 10,
+	}))
+
+	body := "{\"jsonrpc\": \"2.0\",\"method\": \"eth_getTransactionReceipt\",\"params\":[\"0x1234567890123456789012345678901234567890123456789012345678901234\"],\"id\":\"test\"}"
```

Comment:
you can consider using go ethereum's eth client to wrap your json rpc calls instead of writing raw json. https://pkg.go.dev/github.com/ethereum/go-ethereum/ethclient#Client.TransactionReceipt

Reply:
I think this client is a bit different from the rpc spec? It returns a Receipt https://pkg.go.dev/github.com/ethereum/go-ethereum@v1.13.3/core/types#Receipt which is missing fields like `from` and `to` https://github.com/ethereum/go-ethereum/blob/v1.13.3/internal/ethapi/api.go#L1820-L1821